src/App.tsx

```diff
@@ -26,7 +33,41 @@ function App() {
       ? state.context.serviceDataMap[state.context.currentSessionId!]?.machine
       : undefined;
   });
-  const authService = useInterpret(authMachine);
+
+  const router = useRouter();
+
+  const routerReplace = useCallback((url: string) => {
+    /**
+     * Apologies for this line of code. The reason this is here
+     * is that XState + React Fast Refresh causes an error:
+     *
+     * Error: Unable to send event to child 'ctx => ctx.sourceRef'
+     * from service 'auth'.
+     *
+     * router.replace causes this in development, but not in prod
+     *
+     * So, we use window.location.href in development (with the /viz
+     * prefix which Next won't automatically) and router.replace in prod
```

Comment:
isn't some part of the sentence missing here? it looks like this "which Next won't automatically" is cut off

Reply:
```suggestion
     * So, we use window.location.href in development (with the /viz
     * prefix which Next won't automatically add) and router.replace in prod
```